File: h to o/htoo.py
compound_in, num = input().split(" ")
compound_out = input()

def get_stock(compound: str):
    stock = {}
    i = 0
    while i < len(compound):
        element = compound[i]
        increase = 1
        j = k = i + 1
        while j < len(compound) and compound[j].isdigit():
            j += 1
        if k != j:
            increase = int(compound[k:j])
        stock[element] = stock.get(element, 0) + increase
        i = j
    return stock

stock_out = get_stock(compound_out) 
stock_in = get_stock(compound_in)

try:
    print(min([stock_in[element] * int(num) // stock_out[element] for element in stock_out]))
except:
    print(0)

File: h to o/htoo.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class htoo {

    public static Map<Character, Integer> getStock(String compound) {
        Map<Character, Integer> stock = new HashMap<>();
        int i = 0;
        while (i < compound.length()) {
            char element = compound.charAt(i);
            int increase = 1;
            int j = i + 1;
            int k = j;
            while (j < compound.length() && Character.isDigit(compound.charAt(j))) {
                j++;
            }
            if (k != j) {
                increase = Integer.parseInt(compound.substring(k, j));
            }
            stock.merge(element, increase, Integer::sum);
            i = j;
        }
        return stock;
    }

    public static void main(String[] args) {

        Scanner scn = new Scanner(System.in);
        String compoundIn = scn.next();
        int num = scn.nextInt();
        String compoundOut = scn.next();
        scn.close();

        Map<Character, Integer> stockIn = getStock(compoundIn);
        Map<Character, Integer> stockOut = getStock(compoundOut);
        int result = Integer.MAX_VALUE;
        try {
            for (char key: stockOut.keySet()) {
                int amount = stockIn.get(key) * num / stockOut.get(key);
                if (amount < result) {
                    result = amount;
                }
            }
            System.out.println(result);
        }
        catch (Exception e) {
            System.out.println(0);
        }

    }
    
}
